pkg/leaderelection/leaderelection.go

```diff
@@ -73,7 +73,7 @@ func New(log logr.Logger, name, namespace string, kubeClient kubernetes.Interfac
 	}
 	e.leaderElectionCfg = leaderelection.LeaderElectionConfig{
 		Lock:            e.lock,
-		ReleaseOnCancel: true,
+		ReleaseOnCancel: false,
```

Comment:
Changes looks good. Are we able to reproduce the issue ?. I think it will great if we can reach out to the user who is facing the issue to test this fix.

Reply:
Not sure how to reproduce the issue but I would say that `ReleaseOnCancel: false` is safer.